OpenRA.Mods.Common/Widgets/Logic/Ingame/LoadIngamePlayerOrObserverUILogic.cs

```diff
@@ -45,9 +44,13 @@ public LoadIngamePlayerOrObserverUILogic(Widget widget, World world)
 
 			world.GameOver += () =>
 			{
-				worldRoot.RemoveChildren();
-				menuRoot.RemoveChildren();
-				Game.LoadWidget(world, "LEAVE_MAP_WIDGET", menuRoot, new WidgetArgs());
+				var w = Ui.Root.GetOrNull("OBSERVER_WIDGETS");
+				if (w == null)
+					w = Ui.Root.GetOrNull("PLAYER_WIDGETS");
+
+				var optionsButton = w == null ? null : w.GetOrNull<MenuButtonWidget>("OPTIONS_BUTTON");
```

Comment:
Do I need a null check here? It's either observer widgets or player widgets... The ingame menu code had null check for options button so I put it here below too. But is there a side panel without options button?